### PR TITLE
DC-431 Flow Handoff from minion to cloud

### PR DESCRIPTION
**Pull request by FrancescoBuc86**
Jira: https://opennms.atlassian.net/browse/DC-431

End to End Test will be added with https://opennms.atlassian.net/browse/DC-489

### CODE REVIEW

File: minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/parser/FlowSinkModule.java [hunk 1]
```diff
@@ -32,17 +32,16 @@
...
 import org.opennms.horizon.shared.ipc.sink.api.AggregationPolicy;
 import org.opennms.horizon.shared.ipc.sink.api.AsyncPolicy;
 import org.opennms.horizon.shared.ipc.sink.api.SinkModule;
-import org.opennms.horizon.shared.ipc.sink.api.UnmarshalException;
+
 
 public class FlowSinkModule implements SinkModule<TelemetryMessage, TelemetryMessageLog> {
 
-    private final IpcIdentity identity;
+    private final org.opennms.horizon.minion.flows.parser.factory.Identity identity;
```
Comment (patrick-schweizer) [inline minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/parser/FlowSinkModule.java:42]: not sure if we need fully qualified package name here?

File: minion/minion-flows/flows-parser/src/main/resources/OSGI-INF/blueprint/blueprint.xml [hunk 2]
```diff
@@ -18,16 +19,31 @@
...
         <argument ref="messageDispatcherFactory" />
-        <argument ref="udpListenerModule"/>
+        <argument ref="flowSinkModule"/>
+    </bean>
+    <bean id="flowSinkModule" class="org.opennms.horizon.minion.flows.parser.FlowSinkModule" >
+        <argument ref="identity" />
+    </bean>
+    <bean id="UnmarshalException" class="org.opennms.horizon.minion.flows.parser.UnmarshalException" />
+    <bean id="identity" class="org.opennms.horizon.minion.flows.parser.factory.IdentityImpl" >
+        <argument value="$[id]" />
```
Comment (patrick-schweizer) [inline minion/minion-flows/flows-parser/src/main/resources/OSGI-INF/blueprint/blueprint.xml:29]: where does the id come from? maybe systemId? Also, shouldn't it be `${id}` ?

File: minion/minion-flows/flows-parser/src/main/resources/OSGI-INF/blueprint/blueprint.xml [hunk 3]
```diff
@@ -18,16 +19,31 @@
...
+        <argument value="$[location]" />
```
Comment (patrick-schweizer) [inline minion/minion-flows/flows-parser/src/main/resources/OSGI-INF/blueprint/blueprint.xml:30]: I am not sure if this works? maybe it should be `${location}` ?

**Review by patrick-schweizer — COMMENTED** [review]
@FrancescoBuc86 I have some questions around the property handling.

File: minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/parser/FlowSinkModule.java (hunk 1 above)
Comment (FrancescoBuc86) [reply minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/parser/FlowSinkModule.java:42]: Ok, changed it

File: minion/minion-flows/flows-parser/src/main/resources/OSGI-INF/blueprint/blueprint.xml (hunk 3 above)
Comment (FrancescoBuc86) [reply minion/minion-flows/flows-parser/src/main/resources/OSGI-INF/blueprint/blueprint.xml:30]: Same as above

**Pull request merged**